### PR TITLE
Barman support

### PR DESCRIPTION
**Pull request by jt-edb**
This PR includes:
- new role for Barman server deployment: `setup_barmanserver`
- new role for configuring Barman on Postgres nodes: `setup_barman`

Local backups are not supported.
Remote backup methods supported are `postgres` and `rsync`.

### CODE REVIEW

File: lookup_plugins/barman_server.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from ansible.inventory.manager import InventoryManager
+from ansible.plugins.lookup import LookupBase
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+
+        myvars = getattr(self._templar, '_available_variables', {})
+        # Inventory hostname
+        ihn = variables['inventory_hostname']
```
Comment (vibhorkumar123) [inline lookup_plugins/barman_server.py:42]: Can we change the variable name to something meaningful?

File: lookup_plugins/barman_server.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+            if hv['private_ip'] != barman_server_private_ip:
+                continue
+
+            return [
+                dict(
+                    node_type='barmanserver',
+                    ansible_host=hv['ansible_host'],
+                    hostname=hv.get('hostname', hv['ansible_hostname']),
+                    private_ip=hv['private_ip'],
```
Comment (vibhorkumar123) [inline lookup_plugins/barman_server.py:71]: Same here.

File: roles/setup_barman/tasks/setup_barman.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  when: >
+    barman_backup_method not in ['postgres', 'rsync']
+
+- name: Check support for Operating System
+  fail:
+    msg: "Operating System = {{ os }} not supported."
+  when: os not in supported_os
+
+- name: Reference pg_type variables
+  include_vars: "../../init_dbserver/vars/{{ pg_type }}.yml"
```
Comment (vibhorkumar123) [inline roles/setup_barman/tasks/setup_barman.yml:23]: Do you think it's a good practice for using relative paths for variables?

File: roles/setup_barman/tasks/setup_barman.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+- name: Reference barman server variables
+  include_vars: "../../setup_barmanserver/defaults/main.yml"
```
Comment (vibhorkumar123) [inline roles/setup_barman/tasks/setup_barman.yml:26]: Same here

File: roles/setup_barman/templates/barman.rsync.template [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[{{ ansible_hostname }}]
```
Comment (vibhorkumar123) [inline roles/setup_barman/templates/barman.rsync.template:1]: Will ansible hostname include the hostname of the target node?

**Review by vibhorkumar123 — CHANGES_REQUESTED** [review]

File: roles/setup_barman/templates/barman.rsync.template (hunk 5 above)
Comment (jt-edb) [reply roles/setup_barman/templates/barman.rsync.template:1]: Yes, in this case, the target node is the current host.

File: lookup_plugins/barman_server.py (hunk 1 above)
Comment (jt-edb) [reply lookup_plugins/barman_server.py]: Done.

File: lookup_plugins/barman_server.py (hunk 2 above)
Comment (jt-edb) [reply lookup_plugins/barman_server.py]: Done.

File: roles/setup_barman/tasks/setup_barman.yml (hunk 3 above)
Comment (jt-edb) [reply roles/setup_barman/tasks/setup_barman.yml]: Done. Replaced by `include_vars: "{{ role_path }}/../<antoher_role>/vars/<...>"`

File: roles/setup_barman/tasks/setup_barman.yml (hunk 4 above)
Comment (jt-edb) [reply roles/setup_barman/tasks/setup_barman.yml]: Done. Please see https://github.com/EnterpriseDB/edb-ansible/pull/95#discussion_r549275312

**Review by vibhorkumar123 — APPROVED** [review]

**Pull request merged**